### PR TITLE
Corgium improvements

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

If corgium is created inside a mob, it will turn the mob into a corgi for a duration depending on the volume created.

## Why It's Good For The Game

you can turn yourself into a corgi.

## Changelog
:cl:
tweak: If corgium reacts in your body, it will turn you into a corgi for some  time.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
He turned himself into a Corgi.

**Comment by ghost** [comment]
funniest shit I have ever seen

**Comment by eeSPee** [comment]
Could /obj/shapeshift_holder in "modules/spells/spell_types/shapeshift.dm" help improve the code?

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -250,6 +250,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define HAND_REPLACEMENT_TRAIT "magic-hand"
 #define HOT_POTATO_TRAIT "hot-potato"
 #define SABRE_SUICIDE_TRAIT "sabre-suicide"
+#define TRAIT_CORGIUM "corgium"
```
Comment (PowerfulBacon) [inline code/__DEFINES/traits.dm]:
```suggestion
#define CORGIUM_TRAIT "corgium"
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -112,6 +112,42 @@
...
+	color = "#ecca7f"
+	taste_description = "dog treats"
+	var/mob/living/simple_animal/pet/dog/corgi/new_corgi
+
+/datum/reagent/corgium/on_mob_metabolize(mob/living/L)
+	. = ..()
+	new_corgi = new(get_turf(L))
+	new_corgi.key = L.key
+	new_corgi.name = L.name
+	ADD_TRAIT(L, TRAIT_NOBREATH, TRAIT_CORGIUM)
```
Comment (PowerfulBacon) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	ADD_TRAIT(L, TRAIT_NOBREATH, CORGIUM_TRAIT)
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -112,6 +112,42 @@
...
+
+/datum/reagent/corgium/on_mob_life(mob/living/carbon/M)
+	. = ..()
+	//If our corgi died :(
+	if(new_corgi.stat)
+		holder.remove_all_type(type)
+
+/datum/reagent/corgium/on_mob_end_metabolize(mob/living/L)
+	. = ..()
+	REMOVE_TRAIT(L, TRAIT_NOBREATH, TRAIT_CORGIUM)
```
Comment (PowerfulBacon) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	REMOVE_TRAIT(L, TRAIT_NOBREATH, CORGIUM_TRAIT)
```

**Comment by eeSPee** [comment]
Pickleium when?

**Comment by ike709** [comment]
Simple mobs metabolize reagents, right?

**Comment by PowerfulBacon** [comment]
> 
> 
> Simple mobs metabolize reagents, right?

Yes I think so, but the reagents don't go into the simplemob. They stay in the original mob, which gets moved inside the corgi. When the reagents of the original mob stop metabolizing the mob gets moved out of the corgi and the corgi gets deleted. The reagents continue to metabolize because they are in the human not the corgi

**Comment by ike709** [comment]
> They stay in the original mob, which gets moved inside the corgi.

I feel like this can cause bad things to happen.

**Comment by PowerfulBacon** [comment]
> > They stay in the original mob, which gets moved inside the corgi.
> 
> I feel like this can cause bad things to happen.

It's how most other things do it

**Comment by ike709** [comment]
YOLO

**Pull request merged**